### ISSUE TITLE
LandmarkBasedTransformInitializer segfaults on empty point lists

### ISSUE THREAD

**Issue by zivy** [SimpleITK/SimpleITK#368]
The following code segfaults (I expected exception):

```
import SimpleITK as sitk
sitk.LandmarkBasedTransformInitializer(sitk.VersorRigid3DTransform(),[],[])
```
Looking at the SimpleITK LandmarkBasedTransformInitializer it appears that the problem is with the sitkSTLVectorToITKPointVector function, we have an overflow [here](https://github.com/SimpleITK/SimpleITK/blob/495f65e7b8e546349a64416a6dd74c23be812c3c/Code/Common/include/sitkTemplateFunctions.h#L72).

Need to add size check before doing anything
```
if( in.size() > 0 ) {
  unsigned int Dimension = itkPointVectorType::value_type::GetPointDimension();

  for( unsigned int i = 0; i <= in.size()- Dimension; i += Dimension )
    {
    typename itkPointVectorType::value_type pt(&in[i]);
    out.push_back(pt);
    }
}
```

**Comment by blowekamp** [comment]
Looks like you have found the cause of your problem, so why don't you submit a patch.

The logic should check that `in.size() >= Dimension` not zero. It may be better to express this as ` for( unsigned int i = 0; i+Dimension <= in.size(); i += Dimension )` to avoid the pesky signed/unsigned issue in the original statement.

**Comment by zivy** [comment]
Posted it on github so that I don't forget. Will address it later.

It was killing me on one of the notebooks. Testing wouldn't complete and I found the issue only after running it manually. The reason being that the kernel dies and restarts.

**Comment by zivy** [comment]
Fixed by 6434f18f883967943d7f291a4b137c3ed205e7c0.
